perceval/components/source.py

```diff
@@ -21,66 +21,129 @@
 # SOFTWARE.
 
 import math
+import numpy as np
 
 from perceval.utils import SVDistribution, StateVector
-from typing import Dict
+from typing import Dict, Literal
 
 
 class Source:
-    r"""Definition of a single photon source. The default source is a perfect source (emits exactly 1 indistinguishable
-    photon each period).
-
-    :param brightness: the brightness of the source defined as the percentage of unique photon generation
-    :param purity: the ratio of time when photon is emitted alone
-    :param purity_model: `"random"` if additional photons are distinguishable, `"indistinguishable"` otherwise
-    :param indistinguishability: indistinguishability parameter as defined by `indistinguishability_model`
-    :param indistinguishability_model: `"homv"` defines indistinguishability as HOM visibility, `"linear"` defines
-        indistinguishability as ratio of indistinguishable photons
-    :param context: gives a local context for source specific features, like `"discernability_tag"`
-    """
     def __init__(self, brightness: float = 1,
-                 purity: float = 1,
-                 purity_model: str = "random",  # Literal["random", "indistinguishable"]
+                 multiphoton_component: float = None,
```

Comment:
Moving this docstring inside `__init__` breaks the documentation (this part won't show up.. Sphinx is grumpy sometimes).
As Source class is referenced in Perceval documentation, this needs to be moved back to the beginning of the class.